universal-application-tool-0.0.1/app/views/applicant/ApplicantProgramBlockEditView.java

```diff
@@ -199,34 +254,67 @@ private Tag renderReviewButton(Params params) {
         .withClasses(ApplicantStyles.BUTTON_REVIEW);
   }
 
-  private Optional<Tag> renderSkipFileUploadButton(Params params) {
-    // Cannot skip required file questions.
-    if (!params.block().getQuestions().get(0).isOptional()) {
-      return Optional.empty();
-    }
-    String skipUrl =
-        routes.ApplicantProgramBlocksController.skipFile(
-                params.applicantId(), params.programId(), params.block().getId(), params.inReview())
-            .url();
-    return Optional.of(
-        a().attr(HREF, skipUrl)
-            .withText(params.messages().at(MessageKey.BUTTON_SKIP_FILEUPLOAD.getKeyName()))
-            .withId("skip-fileupload-button")
-            .withClasses(ApplicantStyles.BUTTON_REVIEW));
-  }
-
   private Tag renderNextButton(Params params) {
     return submitButton(params.messages().at(MessageKey.BUTTON_NEXT_BLOCK.getKeyName()))
         .withClasses(ApplicantStyles.BUTTON_BLOCK_NEXT)
         .withId("cf-block-submit");
   }
 
+  private Optional<Tag> renderContinueButton(Params params) {
+    if (!hasUploadedFile(params)) {
+      return Optional.empty();
+    }
+    Tag button =
+        submitButton(params.messages().at(MessageKey.BUTTON_KEEP_FILE.getKeyName()))
+            .attr(FORM, FILEUPLOAD_CONTINUE_FORM_ID)
+            .withClasses(ApplicantStyles.BUTTON_BLOCK_NEXT)
+            .withId("fileupload-continue-button");
+    return Optional.of(button);
+  }
+
+  private Optional<Tag> renderDeleteButton(Params params) {
```

Comment:
(blocking) This method returns a delete button or a skip button... a comment with a more accurate name please!

Reply:
Done. PTAL